Framework/Kernel/src/TimeROI.cpp

```diff
@@ -207,7 +207,30 @@ bool TimeROI::isCompletelyInROI(const Types::Core::DateAndTime &startTime,
     return false;
 
   // the value at the start time should be "use"
-  return this->valueAtTime(startTime);
+  return this->valueAtTime(startTime) == ROI_USE;
+}
+
+/**
+ * This method returns true if the entire region between startTime and stopTime is inside an existing ignore interval.
+ * If part of the supplied region is not covered this returns false.
+ */
+bool TimeROI::isCompletelyInMask(const Types::Core::DateAndTime &startTime,
+                                 const Types::Core::DateAndTime &stopTime) const {
+  if (this->empty())
+    return true;
+  if (startTime > m_roi.back())
+    return true;
+  if (stopTime < m_roi.front())
```

Comment:
Only one of these two comparisons can be inclusive. You are correct that having `startTime` of the mask equal the back is correct. However, the `valueAt(m_roi.front())` is actually `ROI_USE`.

Reply:
I assumed the pair `startTime, stopTime` encompass the _open-ended_ interval `[startTime, stopTime)` instead of the closed interval `[startTime, stopTime]`. Which one of the two is the correct one when evaluating `isCompletelyInMask(startTime, stopTime)`?